### PR TITLE
Inc 16/tabs

### PR DESCRIPTION
**Pull request by ilich01**
UPD: 1.Implemented a tab on the profile settings page, where you can also see an example of use.
2. To use Tabs, the "Tab" component is called, which is assigned value = content url and label = tab text. Next, Tabs.Component is used, which displays the required content component.
3. If it is necessary to disable the tab, the "disabled" property is passed to it.
Created a Tabs element for ui-kit. To use it, you need to declare an array, in which the value is passed first (used for navigating through pages by url), the second value is titles (the text that is displayed on the button), the third (if any) is the disabled property. The "Tab" component is called to use
image

image

### CODE REVIEW

File: @ui/ui-kit/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    setActiveTab(router.query.tab as string)
+  }, [router.query.tab])
+
+  const buttonClasses = (tabData: TabData) =>
+    classNames(styles.tab, {
+      [styles[`tab${currentColor}`]]: Boolean(currentColor),
+      [styles.disabled]: tabData.disabled,
+    })
+
+  return (
```
Comment (FightersForJustice) [inline @ui/ui-kit/Tabs/Tabs.tsx:42]: <Tabs props/>

File: @ui/ui-kit/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  label: string
+  disabled?: boolean
+}
+
+interface TabsProps {
+  values: TabData[]
+  color?: (typeof TAB_COLORS)[keyof typeof TAB_COLORS]
+}
+
+const Tab: React.FC<TabsProps> = ({ values, color = TAB_COLORS.PRIMARY }) => {
```
Comment (FightersForJustice) [inline @ui/ui-kit/Tabs/Tabs.tsx:19]: TabWrapper

File: @ui/ui-kit/Tabs/constants.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const TAB_COLORS = Object.freeze({
+  PRIMARY: 'Primary',
+  GHOST: 'Ghost',
+})
```
Comment (FightersForJustice) [inline @ui/ui-kit/Tabs/constants.js:4]: TabsData

File: styles/_library.scss
Comment (FightersForJustice) [inline styles/_library.scss:1]: remove

File: styles/_mixin.scss
Comment (FightersForJustice) [inline styles/_mixin.scss:1]: remove

File: @ui/ui-kit/Tabs/Tabs.module.scss
Comment (FightersForJustice) [inline @ui/ui-kit/Tabs/Tabs.module.scss:1]: tab width 100%

**Comment by TURP1** [comment]
Don't forget about translation

File: @ui/ui-kit/Tabs/Tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import * as Tabs from '@radix-ui/react-tabs'
+import { TAB_COLORS } from './constants'
+import styles from './Tabs.module.scss'
+import classNames from 'classnames'
+
+interface TabData {
+  value: string
```
Comment (irinrudko) [inline @ui/ui-kit/Tabs/Tabs.tsx:9]: isn't it better to name them 'urlValue/url' and title? just a suggestion

**Review by irinrudko — APPROVED** [review]

**Review by maskpa100 — APPROVED** [review]

**Review by TURP1 — APPROVED** [review]

**Review by TURP1 — APPROVED** [review]

**Review by Viarcinski — APPROVED** [review]

**Review by irinrudko — APPROVED** [review]

**Pull request merged**